dnf-behave-tests/features/microdnf/install-xml-base.feature

```diff
@@ -0,0 +1,73 @@
+@no_installroot
+@desctructive
+Feature: microdnf install command on packages with xml:base
+
+@bz1734350
+Scenario: Install packages from local repodata with local packages
+#1. local repo with local packages
+Given I use repository "dnf-ci-fedora"
+ When I execute microdnf with args "install setup"
+ Then the exit code is 0
+  And stderr is empty
+
+
+@bz1734350
+Scenario: Install packages from local repodata with local xml:base
+#2. local repo with local packages (different package location specified using xml:base)
+Given I copy repository "dnf-ci-fedora" for modification
+  And I use repository "dnf-ci-fedora"
+  And I execute "createrepo_c --baseurl file://{context.dnf.installroot}/xml_base/dnf-ci-fedora /{context.dnf.repos[dnf-ci-fedora].path}"
+  And I copy directory "{context.dnf.repos[dnf-ci-fedora].path}" to "/xml_base/dnf-ci-fedora"
+ When I execute microdnf with args "install setup"
```

Comment:
You are repeatedly installing the same package, "setup". It is important to note that since the microdnf tests don't run in an installroot, and since all tests from a single feature file are run in the same container, you are modifying an environment shared between the tests, therefore the `setup` package is already installed in this test (unless you specifically run this single scenario, in which case it won't be installed...).

This is actually seemingly subtle, but quite a big problem. It can hit you in ways you don't expect, it's hard to write tests in such environment and they are much less robust. @m-blaha is looking into how to run each `@no_installroot` scenario in a different container (even though it will have a performance impact) I think, not sure what's the status? @m-blaha will you have a solution soon or do we still need to worry about this?

Anyway, I think if you want to keep these tests in a single feature file, you should pick a different package to install in each scenario. That way they _shouldn't_ be influencing each other (much). An alternative is to split the scenarios into more feature files, that way they will be run in separate containers. In an extreme you could put one scenario per feature file to ensure they don't influence each other.